diff -ur tspc-advanced-o/platform/linux/tsp_local.c tspc-advanced/platform/linux/tsp_local.c
--- tspc-advanced-o/platform/linux/tsp_local.c	2006-11-29 10:06:38.000000000 -0700
+++ tspc-advanced/platform/linux/tsp_local.c	2007-04-27 17:35:38.000000000 -0600
@@ -56,7 +56,7 @@
 char *ScriptInterpretor = "/bin/sh";
 char *ScriptExtension = "sh";
 char *ScriptDir = NULL;
-char *TspHomeDir = "/usr/local/etc/gw6";
+char *TspHomeDir = "/etc/gw6";
 char DirSeparator = '/';
 
 int RootUid = 0;
diff -ur tspc-advanced-o/template/linux.sh tspc-advanced/template/linux.sh
--- tspc-advanced-o/template/linux.sh	2006-11-29 10:06:41.000000000 -0700
+++ tspc-advanced/template/linux.sh	2007-04-27 17:35:38.000000000 -0600
@@ -76,7 +76,6 @@
 Display 1 "--- Start of configuration script. ---"
 Display 1 "Script: " `basename $0`
 
-ifconfig=/sbin/ifconfig
 route=/sbin/route
 ipconfig=/sbin/ip
 rtadvd=/usr/sbin/radvd
@@ -107,33 +106,29 @@
    Display 1 "$TSP_TUNNEL_INTERFACE setup"
    if [ X"${TSP_TUNNEL_MODE}" = X"v6v4" ]; then
       Display 1 "Setting up link to $TSP_SERVER_ADDRESS_IPV4"
-      if [ -x $ipconfig ]; then
 	 ExecNoCheck $ipconfig tunnel del $TSP_TUNNEL_INTERFACE
 	 ExecNoCheck sleep 1
          Exec $ipconfig tunnel add $TSP_TUNNEL_INTERFACE mode sit ttl 64 remote $TSP_SERVER_ADDRESS_IPV4
-      else
-         Exec $ifconfig $TSP_TUNNEL_INTERFACE tunnel ::$TSP_SERVER_ADDRESS_IPV4
-      fi
    fi
 
-   Exec $ifconfig $TSP_TUNNEL_INTERFACE up
+   Exec $ipconfig link set $TSP_TUNNEL_INTERFACE up
 
    PREF=`echo $TSP_CLIENT_ADDRESS_IPV6 | sed "s/:0*/:/g" |cut -d : -f1-2`
-   OLDADDR=`$ifconfig $TSP_TUNNEL_INTERFACE | grep "inet6.* $PREF" | sed -e "s/^.*inet6 addr: //" -e "s/ Scope.*\$//"`
+   OLDADDR=`$ipconfig addr show dev $TSP_TUNNEL_INTERFACE | grep "inet6.* $PREF" | sed -e "s/^.*inet6 addr: //" -e "s/ Scope.*\$//"`
    if [ ! -z $OLDADDR ]; then
       Display 1 "Removing old IPv6 address $OLDADDR"
-      Exec $ifconfig $TSP_TUNNEL_INTERFACE inet6 del $OLDADDR
+      Exec $ipconfig addr del $OLDADDR dev $TSP_TUNNEL_INTERFACE
    fi
    Display 1 "This host is: $TSP_CLIENT_ADDRESS_IPV6/$TSP_TUNNEL_PREFIXLEN"
-   Exec $ifconfig $TSP_TUNNEL_INTERFACE add $TSP_CLIENT_ADDRESS_IPV6/$TSP_TUNNEL_PREFIXLEN
-   Exec $ifconfig $TSP_TUNNEL_INTERFACE mtu 1280
+   Exec $ipconfig addr add $TSP_CLIENT_ADDRESS_IPV6/$TSP_TUNNEL_PREFIXLEN dev $TSP_TUNNEL_INTERFACE 
+   Exec $ipconfig link set $TSP_TUNNEL_INTERFACE mtu 1280
    # 
    # Default route  
    Display 1 "Adding default route"
-   ExecNoCheck $route -A inet6 del ::/0 2>/dev/null # delete old default route
-   ExecNoCheck $route -A inet6 del 2000::/3 2>/dev/null  # delete old gw route
-   Exec $route -A inet6 add ::/0 dev $TSP_TUNNEL_INTERFACE
-   Exec $route -A inet6 add 2000::/3 dev $TSP_TUNNEL_INTERFACE
+   ExecNoCheck $ipconfig -6 route add del ::/0 2>/dev/null # delete old default route
+   ExecNoCheck $ipconfig -6 route del 2000::/3 2>/dev/null  # delete old gw route
+   Exec $ipconfig -6 route add ::/0 dev $TSP_TUNNEL_INTERFACE
+   Exec $ipconfig -6 route add 2000::/3 dev $TSP_TUNNEL_INTERFACE
 fi
 
 # Router configuration if required
@@ -146,12 +141,12 @@
    fi
    Exec $sysctl -w net.ipv6.conf.all.forwarding=1 # ipv6_forwarding enabled
    Display 1 "Adding prefix to $TSP_HOME_INTERFACE"
-   OLDADDR=`$ifconfig $TSP_HOME_INTERFACE | grep "inet6.* $PREF" | sed -e "s/^.*inet6 addr: //" -e "s/ Scope.*\$//"`
+   OLDADDR=`$ipconfig addr show dev $TSP_HOME_INTERFACE | grep "inet6.* $PREF" | sed -e "s/^.*inet6 addr: //" -e "s/ Scope.*\$//"`
    if [ ! -z $OLDADDR ]; then
       Display 1 "Removing old IPv6 address $OLDADDR"
-      Exec $ifconfig $TSP_HOME_INTERFACE inet6 del $OLDADDR
+      Exec $ipconfig addr del $OLDADDR dev $TSP_HOME_INTERFACE
    fi
-   Exec $ifconfig $TSP_HOME_INTERFACE add $TSP_PREFIX::1/64
+   Exec $ipconfig addr add $TSP_PREFIX::1/64 dev $TSP_HOME_INTERFACE
    # Router advertisement configuration 
    Display 1 "Create new $rtadvdconfigfile"
    echo "##### rtadvd.conf made by Gateway6 Client ####" > "$rtadvdconfigfile"
