diff -ur freenet6-client-1.0-orig/template/linux.sh freenet6-client-1.0/template/linux.sh
--- freenet6-client-1.0-orig/template/linux.sh	2003-09-12 16:49:00.000000000 -0600
+++ freenet6-client-1.0/template/linux.sh	2003-12-27 01:09:28.789644574 -0700
@@ -71,8 +71,8 @@
 ipconfig=/sbin/ip
 rtadvd=/usr/sbin/radvd
 sysctl=/sbin/sysctl
-rtadvdconfigfilename=tsprtadvd.conf
-rtadvdconfigfile=$TSP_HOME_DIR/$rtadvdconfigfilename
+rtadvdconfigfilename=radvd.conf
+rtadvdconfigfile=/etc/radvd.conf
 
 if [ -z $TSP_HOME_DIR ]; then
    echo "TSP_HOME_DIR variable not specified!;"
@@ -152,11 +152,12 @@
    echo " };" >> "$rtadvdconfigfile"
    echo "};" >> "$rtadvdconfigfile"
    echo "" >> "$rtadvdconfigfile"
-   /etc/init.d/radvd stop
-   if [ -f $rtadvdconfigfile ]; then
-      KillProcess $rtadvdconfigfile
-      Exec $rtadvd -u radvd -C $rtadvdconfigfile
-      Display 1 "Starting radvd: $rtadvd -u radvd -C $rtadvdconfigfile"
+   if [ -x /var/lock/subsys/radvd ]; then
+	   /etc/rc.d/init.d/radvd restart
+   else
+	   echo 'run "/etc/rc.d/init.d/radvd start" to start radvd'
+   fi
+
    else
       echo "Error : file $rtadvdconfigfile not found"
       exit 1
